quodlibet/quodlibet/ext/events/synchronize_to_device.py

```diff
@@ -0,0 +1,168 @@
+                            enabled_queries.append(query)
+
+                    selected_songs = []
+                    for song in app.library.itervalues():
+                        if any(query.search(song) for query in
+                               enabled_queries):
+                            selected_songs.append(song)
+
+                    filename_list = []
+                    destination_path = config.get('plugins', self.c_path)
+                    try:
+                        for song in selected_songs:
+                            if not self.running:
+                                append("Stopped the synchronization.")
+                                return
+
+                            # prevent the application from becoming unreponsive
+                            while Gtk.events_pending():
+                                Gtk.main_iteration()
+
+                            song_path = song['~filename']
+                            song_file_name = "{} - {} - {} - {}.mp3".format(
+                                song("genre"), song("artist"),
+                                song("album"), song("title"))
+                            valid_chars = "-_.() %s%s" % (
+                                string.ascii_letters, string.digits)
+                            song_file_name = "".join(
+                                char for char in song_file_name if
+                                char in valid_chars)
+                            filename_list.append(song_file_name)
+                            dest_file = destination_path + song_file_name
+                            # skip existing files
+                            if os.path.exists(dest_file):
+                                append(
+                                    "Skipped '{}' because it "
+                                    "already exists.".format(song_file_name))
+                            else:
+                                append(
+                                    "Writing '{}'...".format(dest_file))
+                                copyfile(song_path, dest_file)
+
+                        # delete files which are not
+                        # in the saved searches anymore
+                        for existing_file_name in os.listdir(destination_path):
+                            if existing_file_name not in filename_list:
+                                append(
+                                    "Deleted '{}'.".format(existing_file_name))
+                                os.remove(
+                                    destination_path + existing_file_name)
+
+                        append("Synchronization finished.")
+                    except FileNotFoundError as e:
+                        append(str(e))
+
+                def start(_):
+                    append("Starting...")
+                    self.running = True
+                    synchronize()
+
+                def stop(_):
```

Comment:
Use `button` or something similar as the first argument instead, as this overrides `_` which we use for translation.